src/box/memtx_tree.cc

```diff
@@ -385,12 +412,15 @@ tree_iterator_next_equal_base(struct iterator *iterator, struct tuple **ret)
 	}
 	struct index *idx = iterator->index;
 	struct space *space = space_by_id(iterator->space_id);
+	struct tuple *nearby_tuple = res == NULL ? NULL : res->tuple;
```

Comment:
What's the point of moving nearby_tuples var?

Reply:
It accesses the res variable (pointer to BPS tree element): it was decided to move all such operations before the “MVCC TRANSACTION MANAGER STORY GARBAGE COLLECTION BOUND” line to get rid of possible garbage collection side effects.

Reply:
Ok, get it